### PR TITLE
Create gsi-addon-env template

### PR DESCRIPTION
**Pull request by DavidHuber-NOAA**
### Summary

Creates a downstream GSI environment of the unified-env environment to add builds of bufr/11.7.0, met/9.1.3, and metplus/3.1.1 to support the GSI and the global workflow.

### Testing

Built on Orion and compiled the GSI, GSI-utils, GSI-monitor, and GFS-utils packages against it.  Also successfully loaded the global workflow modules.

### Applications affected

GSI, GSI-monitor, GSI-utils, GFS-utils, and global-workflow.

### Systems affected

Hera, Orion, Hercules, Jet, Gaea, and S4

### Dependencies

None

### Issue(s) addressed

#864

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.

### CODE REVIEW

File: configs/templates/gsi-addon-dev/spack.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  concretizer:
+    unify: when_possible
+
+  view: false
+  include:
+  - site
+  - common
+  definitions:
+  # Note: Set 'compilers' manually; must match upstream list
+  - compilers: []
```
Comment (climbfuji) [inline configs/templates/gsi-addon-dev/spack.yaml:16]: Maybe in a future release of spack-stack, we can automate this with the `spack-stack create env --upstream` command? Infer the compiler list from the first upstream repo provided and set it here.

File: configs/templates/gsi-addon-dev/spack.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  specs:
+  - matrix:
+    - [$packages]
+    - [$compilers]
+
+  # Note: Set 'install_tree' to the upstream unified-env install path
+  upstreams:
+    spack-stack-unified-env:
+       install_tree: /path/to/envs/unified-env/install
```
Comment (climbfuji) [inline configs/templates/gsi-addon-dev/spack.yaml]: This should probably come from the `spack-stack create env --upstream` command. I will test this PR when I create the release candidate for spack-stack-1.6.0 on Orion or Hercules (most likely the latter).

Comment (climbfuji) [reply configs/templates/gsi-addon-dev/spack.yaml]: I confirmed that by using `spack stack create env --template=gsi-addon --upstream=/whatever/your/path/to/envs/unified-env/install --site=pick-your-site` creates lines 22-25, so please remove those

**Review by climbfuji — CHANGES_REQUESTED** [review]

File: configs/templates/gsi-addon-dev/spack.yaml (hunk 2 above)
Comment (climbfuji) [reply configs/templates/gsi-addon-dev/spack.yaml]: And, if you like, leave a comment there that explains that upstreams will be added with the `--upstream` option for `spack stack create env`? But not absolutely necessary, as you wish.

Comment (DavidHuber-NOAA) [reply configs/templates/gsi-addon-dev/spack.yaml]: Fixed.  Thanks for checking!

**Review by climbfuji — APPROVED** [review]
Thanks! I tested this template on my mac and it works fine.

**Review by ulmononian — APPROVED** [review]
looks good! thanks for doing this, @DavidHuber-NOAA!

**Pull request merged**